lib/src/tree.rs

```diff
@@ -121,6 +121,10 @@ impl Tree {
         &self.id
     }
 
+    pub fn legacy_id(&self) -> MergedTreeId {
+        MergedTreeId::Legacy(self.id.clone())
+    }
```

Comment:
nit: I find `tree.legacy_id()` is confusing in this context since it's up to caller whether a `Tree` belongs to legacy tree or merged tree. 

That said, I don't think it would be worth the effort to find and rename to a better name. I assume this function will be inlined back to `MergedTree::id()` once we migrate most API surface to non-legacy API.

Reply:
I agree. I've found it confusing myself. And yes, I'll try to get rid of this pretty soon.